erp_facade_service_openApi.yml

```diff
@@ -0,0 +1,210 @@
+      security:
+        - ApiKeyAuth: []
+      operationId: sapCallback
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/correlationIdRequest'
+      responses:
+        "200":
+          description: OK 
+          headers:
+            X-Correlation-ID:
+              description: GUID for the request for logging/tracing
+              schema:
+                type: string
+        "401":
+          description: Unauthorised - your key is not recognised.
+          headers:
+            X-Correlation-ID:
+              description: GUID for the request for logging/tracing
+              schema:
+                type: string
+          content: {}
+        "404":
+          description: Correlation ID not found.
+          headers:
+            X-Correlation-ID:
+              description: GUID for the request for logging/tracing
+              schema:
+                type: string
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/errorDescription'
+              example:
+                correlationId: 6c254109-8a0f-4446-8649-721775ba9de2
+                errors:
+                - source: erpfacade
+                  description: Correlation ID Not Found.
+        "500":
+          description: Internal Server Error.
+          headers:
+            X-Correlation-ID:
+              description: GUID for the request for logging/tracing
+              schema:
+                type: string
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/exceptionDescription'
+components:
+  schemas:
+    correlationIdRequest:
+      type: object
+      properties:
+        correlationId:
+          type: string
+          description: Correlation Id must be a string.
+    exceptionDescription:
+      type: object
+      properties:
+        correlationId:
+          type: string
+    errorDescription:
+      type: object
+      properties:
+        correlationId:
+          type: string
+        errors:
+          type: array
+          items:
+            $ref: '#/components/schemas/fieldError'
+    fieldError:
+      type: object
+      properties:
+        source:
+          type: string
+        description:
+          type: string
+    eventSchema:
+      type: object
+      properties:
+        specversion:
+          type: string
+        type:
+          type: string
+        source:
+          type: string
+        id:
+          type: string
+        time:
+          type: string
+        subject:
+          type: string
+        datacontenttype:
+          type: string
+        data:
+          type: object
+          properties:
+            property1:
+              type: string
+            property2:
+              type: string
+              format: string
```

Comment:
```suggestion
data:
          type: object
          description: The event payload. This specification does not place any restriction on the type of this information.
```

Reply:
Updated as suggested in different PR https://github.com/UKHO/erp-facade/pull/239. Thanks